process/smartContract/process.go

```diff
@@ -1358,6 +1378,12 @@ func (sc *scProcessor) createSmartContractResults(
 			result.OriginalSender = tx.GetSndAddr()
 		}
 
+		isAsyncTransferBackToSender := callType == vmcommon.AsynchronousCall && bytes.Equal(outAcc.Address, tx.GetSndAddr())
+		isLastOutTransfer := i == lenOutTransfers-1
+		if isLastOutTransfer && isAsyncTransferBackToSender {
```

Comment:
It is our standard. That is how we say this will work.

Reply:
If we define that the last transfer is sent back in the asyncCallback than everyone codes like that.

Reply:
if that is our defined rule in the code -> it means it is enforced.

Reply:
It will work as it is, and it's safe enough - tests will fail if we accidentally break this rule.

But if there is a stricter criterion possible (e.g. multiple conditions on the fields of `outAcc`), we should use that.